src/uart.rs

```diff
@@ -458,14 +593,32 @@ impl<'d> UartDriver<'d> {
         rts: Option<impl Peripheral<P = impl OutputPin> + 'd>,
         config: &config::Config,
     ) -> Result<Self, EspError> {
-        new_common(uart, Some(tx), Some(rx), cts, rts, config)?;
+        let mut q_handle_raw = ptr::null_mut();
+        let q_handle = if config.queue_size > 0 {
+            Some(&mut q_handle_raw)
+        } else {
+            None
+        };
+        new_common(uart, Some(tx), Some(rx), cts, rts, config, q_handle)?;
 
         Ok(Self {
             port: UART::port() as _,
+            queue: q_handle_raw,
             _p: PhantomData,
         })
     }
 
+    /// Retrieves the event queue for this UART. Returns `None` if
+    /// the config specified 0 for `queue_size`.
+    pub fn event_queue(&self) -> Option<Queue<uart_event_t>> {
```

Comment:
Should return `Option<&Queue<...>>>`, as the driver would own the `Queue` instance, even if the Rust code does not own the raw queue_t pointer.